### ISSUE TITLE
Retroarch 1.8.4 for PS3 freezes during any attempt at scanning for rom's. 

### ISSUE THREAD

**Issue by zombiebrains** [libretro/RetroArch#10091]
Retroarch 1.8.4 for PS3 (newest jan 2020 version from brewology) on CFW 4.5.1 Rebug Lite (cobra 8.2 i think).

First off thanks for reading this and your time. 

So Retroarch PS3 freezes during any attempt at scanning for rom's. If I use the scan tool and point it to the main directory it starts to scan the first file and freezes the system with 0% showing at the bottom left and the file name of the first rom. It does the same if I point it to the game system folder the roms are in (eg Sega - Genesis), and again even if I scan for a single file and point it directly to a game rom. By freeze I mean the screen freezes, all movement stops, buttons are all unresponsive and I must shut down the PS3 manually. I can load the roms through load content and they work fine but when I try to scan the same rom I just played, it does not work. The only way i found to not freeze it is to do a manual scan. It does not freeze but it does go to 0% and does not progress further than that. I have attempted to scan working  roms from a usb stick and the PS3 hard drive as well to rule out a usb error of some kind. I have also uninstalled RetroArch and reinstalled it. Then updated the data base and info files before trying again but I get the same issue. My folder structure on my usb is GAME ROMS/ (then sep folders for each system eg) Nintendo - Nintendo Entertainment System/ (then the files). I've spent hours reading through forums with no luck. Should I try to install an older version? It seems like it hangs when first going to query the database, maybe there it a way to point it in the right direction? Or a tool that I could use to scan and make a playlist and manually place it into a specific folder on the PS3 RetroArch directory.

Any help at all would be greatly appreciated.

**Comment by i30817** [comment]
Try to vary the setting 'threaded tasks' in settings->user interface after activating 'show advanced settings'.

Rationale: it sounds like a problem of a thread fucking up on the scan.
Possible consequence: your savestates might become extremely slow.

**Comment by zombiebrains** [comment]
Thanks for the help. I really appreciate it. I will try that now.

**Comment by zombiebrains** [comment]
I tried -threaded tasks- toggled on and toggled off. Unfortunately I get the same result. Do I need to save or apply the toggle change? (it resets to on after reboot). A regular scan of any kind locks up the screen (background movement stops) forcing a hard off, and If I search manually it does not freeze the app but the search goes nowhere. This is the case even if I try to scan for a single rom that I put in the usb root and in a folder on the hdd of the PS3. Any other suggestions? 

Thanks again

**Comment by i30817** [comment]
Nope.

**Comment by floppyD** [comment]
Same here.

**Comment by i30817** [comment]
Uh, btw, you aren't trying to scan or run games on a zip or a rar or 7z btw? Compressed formats are total crap for RA and imo, should even be supported - a opinion that is unpopular, but avoids problems like this...

**Comment by floppyD** [comment]
Not him but there's just 4 .sfc files on the directory I'm trying to scan. The crash only happens at the first time you want to run that ROM or when you try to scan for files, after that you can open RA again and find the ROM you tried to load on recents and launch it just fine.

**Comment by i30817** [comment]
Yeah, that's weird. I don't know how people debug the ps3 - or have one - otherwise i'd try to help you; because 4 bare .sfc files shouldn't provoke a crash.

**Comment by zombiebrains** [comment]
Thanks again,

I have unpacked all of the files with WinRAR except for the arcade mame files. I have tried to scan my super nintendo folder which has only unpacked SFC files in it. It still freezes. I have tried to put a single unpacked NES file in a folder all by itself, one I loaded and played just before so I know it's a working file, and when I scan that folder it still freezes. Must be a ps3 thing.

**Comment by SummonedEgar** [comment]
If you are still experiencing issues you can roll back to any version prior to 1.7.8, scanning works flawlessly, I'm on 1.7.7 right now and it's scanning my folders without any problem

**Comment by floppyD** [comment]
@SummonedEgar 
Was 1.7.8 the build that introduced this issue? I was using some old version and it was working just fine. After updating I started having issues with scanning and shaders. The shader thing I already managed to fix with the help of someone in another issue, though.

**Comment by SummonedEgar** [comment]
I've found this thread on reddit, when trying to figure out why scanning froze my ps3, I installed 1.7.7 and voilà it worked. I haven't read the code, but it seems that 1.7.8 introduced this issue.
https://www.reddit.com/r/PS3/comments/dko5le/retroarch_178_and_up_crashes_during_rom_scanning/
Here you can find all previous versions of Retroarch for PS3:
https://xbins.org/libretro/stable/

**Comment by floppyD** [comment]
Thanks, I think I'll keep this version installed until they fix these issues.

**Comment by zombiebrains** [comment]
I installed some older versions but the interface was not as slick. Thanks for the tip. I'll take a look at 1.7.7

**Comment by IllMethods** [comment]
Anyone tried downgrading for the scans then upgrading once all the playlists are made?

**Comment by kuczi55** [comment]
Yeah, playlists are made after update, but you cannot scan new files (after update of course)

**Comment by ZyHead** [comment]
> Try to vary the setting 'threaded tasks' in settings->user interface after activating 'show advanced settings'.
> 
> Rationale: it sounds like a problem of a thread fucking up on the scan.
> Possible consequence: your savestates might become extremely slow.

I did this and it works ! Thanks Man.
Just reload retroarch when you change the setting to save it.

**Comment by SiegSama** [comment]
> Try to vary the setting 'threaded tasks' in settings->user interface after activating 'show advanced settings'.
> 
> Rationale: it sounds like a problem of a thread fucking up on the scan.
> Possible consequence: your savestates might become extremely slow.

Indeed it worked on my dual nand 4.85 cfw phat. RA version 1.8.5, thanks!

**Comment by Daguri** [comment]
> Try to vary the setting 'threaded tasks' in settings->user interface after activating 'show advanced settings'.
> 
> Rationale: it sounds like a problem of a thread fucking up on the scan.
> Possible consequence: your savestates might become extremely slow.

Thanks so much! Still works with - HFW v4.86.1. > RA v1.8.5. > MultiMan v04.85.01

**Comment by bfalese-navent** [comment]
Worked on 1.8.8 HFW 4.86.1. Disabled threaded tasks.
Thanks!!

**Comment by i30817** [comment]
Seems like two different problems for multiple people.

**Comment by bfalese-navent** [comment]
I also switched from ozone to xmb as default UI.

**Comment by TunebaGamer** [comment]
Hello All
Can someone help me ?

I have a Fat PS3 and these days I install RetroArch ver. 1.9.0. (get in store.brewology.com, CEX ver.) but sometimes, in menu screen when choosing an option, all crashes (RetroArch and PS3 system) then I have to hold PS3 power button to restart system.

I try RetroArch Community Edition today, but the same problem. 
I'm playing SMW happy and alone, then when I'll "close content" to choose Super Bomberman, puff, the system freeze/crash.

During game, never freezes or crashes, only in Menu Screens, when I trying to 'close content", or changing button configuration, or take a look at RetroAchievements list and other options. 
I dont try to scan folders because I dont use this feature, I open game by game.

Please, Please, someone help me!... It's sad :(

**Comment by winters987** [comment]
I just want to confirm that what i30817 said worked for me. My retroarch was freezing when scanning trying to import my files. My freeze was varying from the first file to about 5 files in. I had to manually restart the console each time. i30817's solution fixed it. So the solution was:

1. Settings > User interface
5. Set show advance settings to yes
6. Back out of User interface menu
7. User interface 
8. Set threaded tasks to off 
9. Main menu > restart retroarch
11. Try your import scan again

This worked 100% for me. Figured I'd update the thread if anyone else is trying to do this currently. Works for everything current as of 6/10/2023.
I used multiman to copy off my games into the downloads folder inside retroarch.

**Comment by i30817** [comment]
One problem with this workaround is that threaded tasks are important to savestates GUI update (because of a bad idea that people on the project are too attached to of showing progress bars even on platforms without threads).

It will slow them down to have that off.

So i'd suggest turning that on again after the scan if you have no problems with anything else.

**Comment by bobg16422-boop** [comment]
What was it after all

**Comment by bobg16422-boop** [comment]
Before I found out this was very very long time ago. I was thinking that it should be in a ROMS folder on the tip of the USB. But there's other ways to do it as well.
